src/jdk.compiler/share/classes/com/sun/tools/javac/comp/TransLiterals.java

```diff
@@ -0,0 +1,344 @@
+            return apply;
+        }
+
+        JCExpression processCall(JCExpression stringTemplate) {
+            MethodSymbol appyMeth = lookupMethod(tree.pos(), names.process,
+                    syms.processorType, List.of(syms.stringTemplateType));
+            JCExpression applySelect = make.Select(processor, appyMeth);
+            JCExpression process = make.Apply(null, applySelect, List.of(stringTemplate))
+                    .setType(syms.objectType);
+            JCTypeCast cast = make.TypeCast(tree.type, process);
+            return cast;
+        }
+
+        JCExpression newStringTemplate() {
+            List<LoadableConstant> staticArgValues = List.nil();
+            List<Type> staticArgsTypes =
+                    List.of(syms.methodHandleLookupType, syms.stringType,
+                            syms.methodTypeType);
+            if (useValuesList) {
+                JCNewArray fragmentArray = make.NewArray(make.Type(syms.stringType),
+                        List.nil(), makeStringList(fragments));
+                fragmentArray.type = new ArrayType(syms.stringType, syms.arrayClass);
+                JCNewArray valuesArray = make.NewArray(make.Type(syms.objectType),
+                        List.nil(), expressions);
+                valuesArray.type = new ArrayType(syms.objectType, syms.arrayClass);
+                return bsmCall(names.process, names.newLargeStringTemplate, syms.stringTemplateType,
+                        List.of(fragmentArray, valuesArray),
+                        List.of(fragmentArray.type, valuesArray.type),
+                        staticArgValues, staticArgsTypes);
+            } else {
+                for (String fragment : fragments) {
+                    staticArgValues = staticArgValues.append(LoadableConstant.String(fragment));
+                    staticArgsTypes = staticArgsTypes.append(syms.stringType);
+                }
+                return bsmCall(names.process, names.newStringTemplate, syms.stringTemplateType,
+                        expressions, expressionTypes, staticArgValues, staticArgsTypes);
+            }
+        }
+
+        JCExpression bsmProcessCall() {
+            List<JCExpression> args = expressions.prepend(processor);
+            List<Type> argTypes = expressionTypes.prepend(processor.type);
+            VarSymbol processorSym = (VarSymbol)TreeInfo.symbol(processor);
+            List<LoadableConstant> staticArgValues = List.of(processorSym.asMethodHandle(true));
+            List<Type> staticArgsTypes =
+                    List.of(syms.methodHandleLookupType, syms.stringType,
+                            syms.methodTypeType, syms.methodHandleType);
+            for (String fragment : fragments) {
+                staticArgValues = staticArgValues.append(LoadableConstant.String(fragment));
+                staticArgsTypes = staticArgsTypes.append(syms.stringType);
+            }
+            return bsmCall(names.process, names.processStringTemplate, tree.type,
```

Comment:
What happens if we have too many fragments here? (e.g. > 200). That case seems handled in the RAW case.

Reply:
Criteria in isLinkageProcessor() prevents this from being called.

```
        boolean isLinkageProcessor() {
            return processor != null &&
                   !useValuesList &&  <=====
                   types.isSubtype(processor.type, syms.linkageType) &&
                   processor.type.isFinal() &&
                   TreeInfo.symbol(processor) instanceof VarSymbol varSymbol &&
                   varSymbol.isStatic() &&
                   varSymbol.isFinal();
        }
```